selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/factories/CreateFactoryPage.java

```diff
@@ -86,95 +104,120 @@ public CreateFactoryPage(SeleniumWebDriver seleniumWebDriver, Loader loader) {
   }
 
   public void clickOnCompleteTemplateButton() {
-    redrawUiElementsTimeout.until(visibilityOf(completeTemplateButton)).click();
+    seleniumWebDriverHelper.waitAndClick(completeTemplateButton);
   }
 
   public void waitTemplateButtons() {
-    redrawUiElementsTimeout.until(visibilityOf(minimalTemplateButton));
-    redrawUiElementsTimeout.until(visibilityOf(completeTemplateButton));
+    seleniumWebDriverHelper.waitVisibility(minimalTemplateButton);
+    seleniumWebDriverHelper.waitVisibility(completeTemplateButton);
   }
 
-  public Boolean isCreateFactoryButtonDisabled() {
+  public Boolean isCreateFactoryButtonEnabled() {
+    WaitUtils.sleepQuietly(1);
     String attrValue =
-        redrawUiElementsTimeout
-            .until(visibilityOf(createFactoryButton))
-            .getAttribute("aria-disabled");
-    return Boolean.parseBoolean(attrValue);
+        seleniumWebDriverHelper.waitVisibility(createFactoryButton).getAttribute("aria-disabled");
+    return !(Boolean.parseBoolean(attrValue));
   }
 
   public void clickOnCreateFactoryButton() {
-    redrawUiElementsTimeout.until(visibilityOf(createFactoryButton)).click();
+    seleniumWebDriverHelper.waitAndClick(createFactoryButton);
   }
 
   public void clickOnWorkspaceFromList(String workspaceName) {
-    redrawUiElementsTimeout
-        .until(
-            visibilityOfElementLocated(
-                By.xpath(format(Locators.WORKSPACE_NAME_XPATH, workspaceName))))
-        .click();
+    seleniumWebDriverHelper.waitAndClick(
+        By.xpath(format(Locators.WORKSPACE_NAME_XPATH, workspaceName)));
   }
 
   public void waitWorkspaceNameInList(String workspaceName) {
-    redrawUiElementsTimeout.until(
-        visibilityOfElementLocated(By.xpath(format(Locators.WORKSPACE_NAME_XPATH, workspaceName))));
+    seleniumWebDriverHelper.waitVisibility(
+        By.xpath(format(Locators.WORKSPACE_NAME_XPATH, workspaceName)));
   }
 
-  public String getErrorMessage() {
-    return redrawUiElementsTimeout
-        .until(visibilityOfElementLocated(By.xpath(Locators.NEW_FACTORY_ERROR_MESSAGE)))
-        .getText();
+  public void waitErrorMessage(String expectedText) {
+    seleniumWebDriverHelper.waitTextEqualsTo(
+        By.xpath(Locators.NEW_FACTORY_ERROR_MESSAGE), expectedText);
   }
 
   public void waitErrorMessageNotVisible() {
-    redrawUiElementsTimeout.until(
-        invisibilityOfElementLocated(By.xpath(Locators.NEW_FACTORY_ERROR_MESSAGE)));
+    seleniumWebDriverHelper.waitInvisibility(By.xpath(Locators.NEW_FACTORY_ERROR_MESSAGE));
   }
 
   public void waitUploadFileButton() {
-    redrawUiElementsTimeout.until(visibilityOf(uploadFileButton));
+    seleniumWebDriverHelper.waitVisibilityAndGetEnableState(uploadFileButton);
   }
 
-  public void waitGitUrlField() {
-    redrawUiElementsTimeout.until(visibilityOfElementLocated(By.id(Locators.GIT_URL_ID)));
+  public void uploadSelectedConfigFile(Path resourcesUploadFile) throws IOException {
```

Comment:
It's better to have short note about why we should use `hiddenUploadInputButton` instead of regular `uploadFileButton` to make the code clear.